common/src/main/scala/org/apache/comet/CometConf.scala

```diff
@@ -383,6 +383,13 @@ object CometConf {
     .booleanConf
     .createWithDefault(false)
 
+  val COMET_SORTMERGEJOIN_CHECK_KEY_TYPES: ConfigEntry[Boolean] =
```

Comment:
thanks @planga82 
I'm not sure if we need to change the conf for the unit test purpose

Reply:
thanks @comphead . Any idea how to do this in a cleaner way? We could move this constant to other place and avoid putting it in CometConf. In that way we avoid to expose a testing configuration. What do you think?

Reply:
Tbh to test variety of supported datatypes I liked the @andygrove approach introduced in https://github.com/apache/datafusion-comet/pull/351, wondering if we can do that

Reply:
I'm going to review it. Thanks!

Reply:
I think it's not the same idea and not applicable here. I'll close this PR and try to think of another way to do it without adding configuration. Thanks again!